Let the embedder build its page clients on top of the empty clients.

The embedder subclasses EmptyChromeClient and EmptyFrameLoaderClient and
overrides only what it implements; their methods become override instead of
final. Behavior of the empty clients is unchanged.

diff --git a/Source/WebCore/loader/EmptyClients.h b/Source/WebCore/loader/EmptyClients.h
index 09a23e6..3d1874d 100644
--- a/Source/WebCore/loader/EmptyClients.h
+++ b/Source/WebCore/loader/EmptyClients.h
@@ -61,174 +61,174 @@ class EmptyChromeClient : public ChromeClient {
 
     void chromeDestroyed() override { }
 
-    void setWindowRect(const FloatRect&) final { }
-    FloatRect windowRect() const final { return FloatRect(); }
+    void setWindowRect(const FloatRect&) override { }
+    FloatRect windowRect() const override { return FloatRect(); }
 
-    FloatRect pageRect() const final { return FloatRect(); }
+    FloatRect pageRect() const override { return FloatRect(); }
 
-    void focus() final { }
-    void unfocus() final { }
+    void focus() override { }
+    void unfocus() override { }
 
-    bool canTakeFocus(FocusDirection) const final { return false; }
-    void takeFocus(FocusDirection) final { }
+    bool canTakeFocus(FocusDirection) const override { return false; }
+    void takeFocus(FocusDirection) override { }
 
-    void focusedElementChanged(Element*, LocalFrame*, FocusOptions, BroadcastFocusedElement) final { }
-    void focusedFrameChanged(Frame*) final { }
+    void focusedElementChanged(Element*, LocalFrame*, FocusOptions, BroadcastFocusedElement) override { }
+    void focusedFrameChanged(Frame*) override { }
 
-    RefPtr<Page> createWindow(LocalFrame&, const String&, const WindowFeatures&, const NavigationAction&) final { return nullptr; }
-    void show() final { }
+    RefPtr<Page> createWindow(LocalFrame&, const String&, const WindowFeatures&, const NavigationAction&) override { return nullptr; }
+    void show() override { }
 
-    bool canRunModal() const final { return false; }
-    void runModal() final { }
+    bool canRunModal() const override { return false; }
+    void runModal() override { }
 
-    bool toolbarsVisible() const final { return false; }
-    bool statusbarVisible() const final { return false; }
-    bool scrollbarsVisible() const final { return false; }
-    bool menubarVisible() const final { return false; }
+    bool toolbarsVisible() const override { return false; }
+    bool statusbarVisible() const override { return false; }
+    bool scrollbarsVisible() const override { return false; }
+    bool menubarVisible() const override { return false; }
 
-    void setResizable(bool) final { }
+    void setResizable(bool) override { }
 
-    void addMessageToConsole(JSC::MessageSource, JSC::MessageLevel, const String&, unsigned, unsigned, const String&) final { }
+    void addMessageToConsole(JSC::MessageSource, JSC::MessageLevel, const String&, unsigned, unsigned, const String&) override { }
 
-    bool canRunBeforeUnloadConfirmPanel() final { return false; }
-    bool runBeforeUnloadConfirmPanel(String&&, LocalFrame&) final { return true; }
+    bool canRunBeforeUnloadConfirmPanel() override { return false; }
+    bool runBeforeUnloadConfirmPanel(String&&, LocalFrame&) override { return true; }
 
-    void closeWindow() final { }
+    void closeWindow() override { }
 
-    void rootFrameAdded(const LocalFrame&) final { }
-    void rootFrameRemoved(const LocalFrame&) final { }
+    void rootFrameAdded(const LocalFrame&) override { }
+    void rootFrameRemoved(const LocalFrame&) override { }
 
-    void runJavaScriptAlert(LocalFrame&, const String&) final { }
-    bool runJavaScriptConfirm(LocalFrame&, const String&) final { return false; }
-    bool runJavaScriptPrompt(LocalFrame&, const String&, const String&, String&) final { return false; }
+    void runJavaScriptAlert(LocalFrame&, const String&) override { }
+    bool runJavaScriptConfirm(LocalFrame&, const String&) override { return false; }
+    bool runJavaScriptPrompt(LocalFrame&, const String&, const String&, String&) override { return false; }
 
-    bool selectItemWritingDirectionIsNatural() final { return false; }
-    bool selectItemAlignmentFollowsMenuWritingDirection() final { return false; }
-    RefPtr<PopupMenu> createPopupMenu(PopupMenuClient&) const final;
-    RefPtr<SearchPopupMenu> createSearchPopupMenu(PopupMenuClient&) const final;
+    bool selectItemWritingDirectionIsNatural() override { return false; }
+    bool selectItemAlignmentFollowsMenuWritingDirection() override { return false; }
+    RefPtr<PopupMenu> createPopupMenu(PopupMenuClient&) const override;
+    RefPtr<SearchPopupMenu> createSearchPopupMenu(PopupMenuClient&) const override;
 
-    KeyboardUIMode keyboardUIMode() final { return KeyboardAccessDefault; }
+    KeyboardUIMode keyboardUIMode() override { return KeyboardAccessDefault; }
 
-    bool hasAccessoryMousePointingDevice() const final { return false; }
-    bool hoverSupportedByPrimaryPointingDevice() const final { return false; };
-    bool hoverSupportedByAnyAvailablePointingDevice() const final { return false; }
-    std::optional<PointerCharacteristics> pointerCharacteristicsOfPrimaryPointingDevice() const final { return std::nullopt; };
-    OptionSet<PointerCharacteristics> pointerCharacteristicsOfAllAvailablePointingDevices() const final { return { }; }
+    bool hasAccessoryMousePointingDevice() const override { return false; }
+    bool hoverSupportedByPrimaryPointingDevice() const override { return false; };
+    bool hoverSupportedByAnyAvailablePointingDevice() const override { return false; }
+    std::optional<PointerCharacteristics> pointerCharacteristicsOfPrimaryPointingDevice() const override { return std::nullopt; };
+    OptionSet<PointerCharacteristics> pointerCharacteristicsOfAllAvailablePointingDevices() const override { return { }; }
 
-    void invalidateRootView(const IntRect&) final { }
+    void invalidateRootView(const IntRect&) override { }
     void invalidateContentsAndRootView(const IntRect&) override { }
-    void invalidateContentsForSlowScroll(const IntRect&) final { }
-    void scroll(const IntSize&, const IntRect&, const IntRect&) final { }
-
-    IntPoint screenToRootView(const IntPoint& p) const final { return p; }
-    IntPoint rootViewToScreen(const IntPoint& p) const final { return p; }
-    IntRect rootViewToScreen(const IntRect& r) const final { return r; }
-    IntPoint accessibilityScreenToRootView(const IntPoint& p) const final { return p; };
-    IntRect rootViewToAccessibilityScreen(const IntRect& r) const final { return r; };
+    void invalidateContentsForSlowScroll(const IntRect&) override { }
+    void scroll(const IntSize&, const IntRect&, const IntRect&) override { }
+
+    IntPoint screenToRootView(const IntPoint& p) const override { return p; }
+    IntPoint rootViewToScreen(const IntPoint& p) const override { return p; }
+    IntRect rootViewToScreen(const IntRect& r) const override { return r; }
+    IntPoint accessibilityScreenToRootView(const IntPoint& p) const override { return p; };
+    IntRect rootViewToAccessibilityScreen(const IntRect& r) const override { return r; };
 #if PLATFORM(IOS_FAMILY)
-    void relayAccessibilityNotification(String&&, RetainPtr<NSData>&&) const final { };
-    void relayAriaNotifyNotification(AriaNotifyData&&) const final { };
-    void relayLiveRegionNotification(LiveRegionAnnouncementData&&) const final { };
+    void relayAccessibilityNotification(String&&, RetainPtr<NSData>&&) const override { };
+    void relayAriaNotifyNotification(AriaNotifyData&&) const override { };
+    void relayLiveRegionNotification(LiveRegionAnnouncementData&&) const override { };
 #endif
 
-    void didFinishLoadingImageForElement(HTMLImageElement&) final { }
+    void didFinishLoadingImageForElement(HTMLImageElement&) override { }
 
-    PlatformPageClient platformPageClient() const final { return 0; }
-    void contentsSizeChanged(LocalFrame&, const IntSize&) const final { }
-    void intrinsicContentsSizeChanged(const IntSize&) const final { }
+    PlatformPageClient platformPageClient() const override { return 0; }
+    void contentsSizeChanged(LocalFrame&, const IntSize&) const override { }
+    void intrinsicContentsSizeChanged(const IntSize&) const override { }
 
-    void mouseDidMoveOverElement(const HitTestResult&, OptionSet<PlatformEventModifier>, const String&, TextDirection) final { }
+    void mouseDidMoveOverElement(const HitTestResult&, OptionSet<PlatformEventModifier>, const String&, TextDirection) override { }
 
-    void print(LocalFrame&, const StringWithDirection&) final { }
+    void print(LocalFrame&, const StringWithDirection&) override { }
 
-    void exceededDatabaseQuota(LocalFrame&, const String&, DatabaseDetails) final { }
+    void exceededDatabaseQuota(LocalFrame&, const String&, DatabaseDetails) override { }
 
-    void reachedMaxAppCacheSize(int64_t) final { }
+    void reachedMaxAppCacheSize(int64_t) override { }
 
-    RefPtr<ColorChooser> createColorChooser(ColorChooserClient&, const Color&) final;
+    RefPtr<ColorChooser> createColorChooser(ColorChooserClient&, const Color&) override;
 
-    RefPtr<DataListSuggestionPicker> createDataListSuggestionPicker(DataListSuggestionsClient&) final;
-    bool canShowDataListSuggestionLabels() const final { return false; }
+    RefPtr<DataListSuggestionPicker> createDataListSuggestionPicker(DataListSuggestionsClient&) override;
+    bool canShowDataListSuggestionLabels() const override { return false; }
 
-    RefPtr<DateTimeChooser> createDateTimeChooser(DateTimeChooserClient&) final;
+    RefPtr<DateTimeChooser> createDateTimeChooser(DateTimeChooserClient&) override;
 
-    void setTextIndicator(RefPtr<TextIndicator>&&) const final;
-    void updateTextIndicator(RefPtr<TextIndicator>&&) const final;
+    void setTextIndicator(RefPtr<TextIndicator>&&) const override;
+    void updateTextIndicator(RefPtr<TextIndicator>&&) const override;
 
-    DisplayRefreshMonitorFactory* displayRefreshMonitorFactory() const final;
+    DisplayRefreshMonitorFactory* displayRefreshMonitorFactory() const override;
 
-    void runOpenPanel(LocalFrame&, FileChooser&) final;
-    void showShareSheet(ShareDataWithParsedURL&&, CompletionHandler<void(bool)>&&) final;
-    void loadIconForFiles(const Vector<String>&, FileIconLoader&) final { }
+    void runOpenPanel(LocalFrame&, FileChooser&) override;
+    void showShareSheet(ShareDataWithParsedURL&&, CompletionHandler<void(bool)>&&) override;
+    void loadIconForFiles(const Vector<String>&, FileIconLoader&) override { }
 
-    void elementDidFocus(Element&, const FocusOptions&) final { }
-    void elementDidBlur(Element&) final { }
+    void elementDidFocus(Element&, const FocusOptions&) override { }
+    void elementDidBlur(Element&) override { }
 
-    void setCursor(const Cursor&) final { }
-    void setCursorHiddenUntilMouseMoves(bool) final { }
+    void setCursor(const Cursor&) override { }
+    void setCursorHiddenUntilMouseMoves(bool) override { }
 
-    void scrollContainingScrollViewsToRevealRect(const IntRect&) const final { }
-    void scrollMainFrameToRevealRect(const IntRect&) const final { }
+    void scrollContainingScrollViewsToRevealRect(const IntRect&) const override { }
+    void scrollMainFrameToRevealRect(const IntRect&) const override { }
 
-    void attachRootGraphicsLayer(LocalFrame&, GraphicsLayer*) final { }
-    void attachViewOverlayGraphicsLayer(GraphicsLayer*) final { }
-    void setNeedsOneShotDrawingSynchronization() final { }
-    void triggerRenderingUpdate() final { }
+    void attachRootGraphicsLayer(LocalFrame&, GraphicsLayer*) override { }
+    void attachViewOverlayGraphicsLayer(GraphicsLayer*) override { }
+    void setNeedsOneShotDrawingSynchronization() override { }
+    void triggerRenderingUpdate() override { }
 
 #if PLATFORM(WIN)
-    void AXStartFrameLoad() final { }
-    void AXFinishFrameLoad() final { }
+    void AXStartFrameLoad() override { }
+    void AXFinishFrameLoad() override { }
 #endif
 
 #if PLATFORM(PLAYSTATION) || PLATFORM(HAIKU)
-    void postAccessibilityNotification(AccessibilityObject&, AXNotification) final { }
-    void postAccessibilityNodeTextChangeNotification(AccessibilityObject*, AXTextChange, unsigned, const String&) final { }
-    void postAccessibilityFrameLoadingEventNotification(AccessibilityObject*, AXLoadingEvent) final { }
+    void postAccessibilityNotification(AccessibilityObject&, AXNotification) override { }
+    void postAccessibilityNodeTextChangeNotification(AccessibilityObject*, AXTextChange, unsigned, const String&) override { }
+    void postAccessibilityFrameLoadingEventNotification(AccessibilityObject*, AXLoadingEvent) override { }
 #endif
 
 #if ENABLE(IOS_TOUCH_EVENTS)
-    void didPreventDefaultForEvent() final { }
+    void didPreventDefaultForEvent() override { }
 #endif
 
 #if PLATFORM(IOS_FAMILY)
-    void didReceiveMobileDocType(bool) final { }
-    void setNeedsScrollNotifications(LocalFrame&, bool) final { }
-    void didFinishContentChangeObserving(LocalFrame&, WKContentChange) final { }
-    void notifyRevealedSelectionByScrollingFrame(LocalFrame&) final { }
-    void didLayout(LayoutType) final { }
-    void didStartOverflowScroll() final { }
-    void didEndOverflowScroll() final { }
+    void didReceiveMobileDocType(bool) override { }
+    void setNeedsScrollNotifications(LocalFrame&, bool) override { }
+    void didFinishContentChangeObserving(LocalFrame&, WKContentChange) override { }
+    void notifyRevealedSelectionByScrollingFrame(LocalFrame&) override { }
+    void didLayout(LayoutType) override { }
+    void didStartOverflowScroll() override { }
+    void didEndOverflowScroll() override { }
 
-    void suppressFormNotifications() final { }
-    void restoreFormNotifications() final { }
+    void suppressFormNotifications() override { }
+    void restoreFormNotifications() override { }
 
-    void addOrUpdateScrollingLayer(Node*, PlatformLayer*, PlatformLayer*, const IntSize&, bool, bool) final { }
-    void removeScrollingLayer(Node*, PlatformLayer*, PlatformLayer*) final { }
+    void addOrUpdateScrollingLayer(Node*, PlatformLayer*, PlatformLayer*, const IntSize&, bool, bool) override { }
+    void removeScrollingLayer(Node*, PlatformLayer*, PlatformLayer*) override { }
 
-    void webAppOrientationsUpdated() final { }
-    void showPlaybackTargetPicker(bool, RouteSharingPolicy, const String&) final { }
+    void webAppOrientationsUpdated() override { }
+    void showPlaybackTargetPicker(bool, RouteSharingPolicy, const String&) override { }
 
-    bool showDataDetectorsUIForElement(const Element&, const Event&) final { return false; }
+    bool showDataDetectorsUIForElement(const Element&, const Event&) override { return false; }
 #endif // PLATFORM(IOS_FAMILY)
 
 #if ENABLE(ORIENTATION_EVENTS)
-    IntDegrees deviceOrientation() const final { return 0; }
+    IntDegrees deviceOrientation() const override { return 0; }
 #endif
 
 #if PLATFORM(IOS_FAMILY)
-    bool isStopping() final { return false; }
+    bool isStopping() override { return false; }
 #endif
 
-    void wheelEventHandlersChanged(bool) final { }
+    void wheelEventHandlersChanged(bool) override { }
     
-    bool isEmptyChromeClient() const final { return true; }
+    bool isEmptyChromeClient() const override { return true; }
 
-    void didAssociateFormControls(const Vector<Ref<Element>>&, LocalFrame&) final { }
-    bool shouldNotifyOnFormChanges() final { return false; }
+    void didAssociateFormControls(const Vector<Ref<Element>>&, LocalFrame&) override { }
+    bool shouldNotifyOnFormChanges() override { return false; }
 
-    RefPtr<Icon> createIconForFiles(const Vector<String>& /* filenames */) final;
+    RefPtr<Icon> createIconForFiles(const Vector<String>& /* filenames */) override;
 
-    void requestCookieConsent(CompletionHandler<void(CookieConsentDecisionResult)>&&) final;
+    void requestCookieConsent(CompletionHandler<void(CookieConsentDecisionResult)>&&) override;
 };
 
 DiagnosticLoggingClient& emptyDiagnosticLoggingClient();
diff --git a/Source/WebCore/loader/EmptyFrameLoaderClient.h b/Source/WebCore/loader/EmptyFrameLoaderClient.h
index 5ebee25..15e13a6 100644
--- a/Source/WebCore/loader/EmptyFrameLoaderClient.h
+++ b/Source/WebCore/loader/EmptyFrameLoaderClient.h
@@ -39,168 +39,168 @@ public:
 private:
     Ref<DocumentLoader> createDocumentLoader(ResourceRequest&&, SubstituteData&&) override;
 
-    bool hasWebView() const final;
+    bool hasWebView() const override;
 
-    void makeRepresentation(DocumentLoader*) final;
+    void makeRepresentation(DocumentLoader*) override;
 
 #if PLATFORM(IOS_FAMILY)
-    bool forceLayoutOnRestoreFromBackForwardCache() final;
+    bool forceLayoutOnRestoreFromBackForwardCache() override;
 #endif
 
-    void forceLayoutForNonHTML() final;
+    void forceLayoutForNonHTML() override;
 
-    void setCopiesOnScroll() final;
+    void setCopiesOnScroll() override;
 
-    void detachedFromParent2() final;
-    void detachedFromParent3() final;
+    void detachedFromParent2() override;
+    void detachedFromParent3() override;
 
-    void convertMainResourceLoadToDownload(DocumentLoader*, const ResourceRequest&, const ResourceResponse&) final;
+    void convertMainResourceLoadToDownload(DocumentLoader*, const ResourceRequest&, const ResourceResponse&) override;
 
-    void assignIdentifierToInitialRequest(ResourceLoaderIdentifier, DocumentLoader*, const ResourceRequest&) final;
+    void assignIdentifierToInitialRequest(ResourceLoaderIdentifier, DocumentLoader*, const ResourceRequest&) override;
     bool shouldUseCredentialStorage(DocumentLoader*, ResourceLoaderIdentifier) override;
-    void dispatchWillSendRequest(DocumentLoader*, ResourceLoaderIdentifier, ResourceRequest&, const ResourceResponse&) final;
-    void dispatchDidReceiveAuthenticationChallenge(DocumentLoader*, ResourceLoaderIdentifier, const AuthenticationChallenge&) final;
+    void dispatchWillSendRequest(DocumentLoader*, ResourceLoaderIdentifier, ResourceRequest&, const ResourceResponse&) override;
+    void dispatchDidReceiveAuthenticationChallenge(DocumentLoader*, ResourceLoaderIdentifier, const AuthenticationChallenge&) override;
 #if USE(PROTECTION_SPACE_AUTH_CALLBACK)
-    bool canAuthenticateAgainstProtectionSpace(DocumentLoader*, ResourceLoaderIdentifier, const ProtectionSpace&) final;
+    bool canAuthenticateAgainstProtectionSpace(DocumentLoader*, ResourceLoaderIdentifier, const ProtectionSpace&) override;
 #endif
 
 #if PLATFORM(IOS_FAMILY)
-    RetainPtr<CFDictionaryRef> connectionProperties(DocumentLoader*, ResourceLoaderIdentifier) final;
+    RetainPtr<CFDictionaryRef> connectionProperties(DocumentLoader*, ResourceLoaderIdentifier) override;
 #endif
 
-    void dispatchDidReceiveResponse(DocumentLoader*, ResourceLoaderIdentifier, const ResourceResponse&) final;
-    void dispatchDidReceiveContentLength(DocumentLoader*, ResourceLoaderIdentifier, int) final;
-    void dispatchDidFinishLoading(DocumentLoader*, ResourceLoaderIdentifier) final;
+    void dispatchDidReceiveResponse(DocumentLoader*, ResourceLoaderIdentifier, const ResourceResponse&) override;
+    void dispatchDidReceiveContentLength(DocumentLoader*, ResourceLoaderIdentifier, int) override;
+    void dispatchDidFinishLoading(DocumentLoader*, ResourceLoaderIdentifier) override;
 #if ENABLE(DATA_DETECTION)
-    void dispatchDidFinishDataDetection(NSArray *) final;
+    void dispatchDidFinishDataDetection(NSArray *) override;
 #endif
-    void dispatchDidFailLoading(DocumentLoader*, ResourceLoaderIdentifier, const ResourceError&) final;
-    bool dispatchDidLoadResourceFromMemoryCache(DocumentLoader*, const ResourceRequest&, const ResourceResponse&, int) final;
-
-    void dispatchDidDispatchOnloadEvents() final;
-    void dispatchDidReceiveServerRedirectForProvisionalLoad() final;
-    void dispatchDidCancelClientRedirect() final;
-    void dispatchWillPerformClientRedirect(const URL&, double, WallTime, LockBackForwardList) final;
-    void dispatchDidChangeLocationWithinPage() final;
-    void dispatchDidPushStateWithinPage() final;
-    void dispatchDidReplaceStateWithinPage() final;
-    void dispatchDidPopStateWithinPage() final;
-    void dispatchWillClose() final;
-    void dispatchDidStartProvisionalLoad() final;
-    void dispatchDidReceiveTitle(const StringWithDirection&) final;
-    void dispatchDidCommitLoad(std::optional<HasInsecureContent>, std::optional<UsedLegacyTLS>, std::optional<WasPrivateRelayed>) final;
-    void dispatchDidFailProvisionalLoad(const ResourceError&, WillContinueLoading, WillInternallyHandleFailure) final;
-    void dispatchDidFailLoad(const ResourceError&) final;
-    void dispatchDidFinishDocumentLoad() final;
-    void dispatchDidFinishLoad() final;
-    void dispatchDidReachLayoutMilestone(OptionSet<LayoutMilestone>) final;
-    void dispatchDidReachVisuallyNonEmptyState() final;
-
-    LocalFrame* dispatchCreatePage(const NavigationAction&, NewFrameOpenerPolicy) final;
-    void dispatchShow() final;
-
-    void dispatchDecidePolicyForResponse(const ResourceResponse&, const ResourceRequest&, const String&, FramePolicyFunction&&) final;
-    void dispatchDecidePolicyForNewWindowAction(const NavigationAction&, const ResourceRequest&, FormState*, const String&, std::optional<HitTestResult>&&, FramePolicyFunction&&) final;
-    void dispatchDecidePolicyForNavigationAction(const NavigationAction&, const ResourceRequest&, const ResourceResponse& redirectResponse, FormState*, const String&, std::optional<NavigationIdentifier>, std::optional<HitTestResult>&&, bool, NavigationUpgradeToHTTPSBehavior, SandboxFlags, PolicyDecisionMode, FramePolicyFunction&&) final;
-    void updateSandboxFlags(SandboxFlags) final;
-    void updateOpener(std::optional<FrameIdentifier>) final;
-    void setPrinting(bool, FloatSize, FloatSize, float, AdjustViewSize) final;
-    void cancelPolicyCheck() final;
-
-    void dispatchUnableToImplementPolicy(const ResourceError&) final;
-
-    void dispatchWillSendSubmitEvent(Ref<FormState>&&) final;
-    void dispatchWillSubmitForm(FormState&, URL&& requestURL, String&& method, CompletionHandler<void()>&&) final;
-
-    void revertToProvisionalState(DocumentLoader*) final;
-    void setMainDocumentError(DocumentLoader*, const ResourceError&) final;
-
-    void setMainFrameDocumentReady(bool) final;
-
-    void startDownload(const ResourceRequest&, const String&, FromDownloadAttribute = FromDownloadAttribute::No) final;
-
-    void willChangeTitle(DocumentLoader*) final;
-    void didChangeTitle(DocumentLoader*) final;
-
-    void willReplaceMultipartContent() final;
-    void didReplaceMultipartContent() final;
-
-    void committedLoad(DocumentLoader*, const SharedBuffer&) final;
-    void finishedLoading(DocumentLoader*) final;
-
-    void loadStorageAccessQuirksIfNeeded() final;
-
-    bool shouldFallBack(const ResourceError&) const final;
-
-    bool canHandleRequest(const ResourceRequest&) const final;
-    bool canShowMIMEType(const String&) const final;
-    bool canShowMIMETypeAsHTML(const String&) const final;
-    bool representationExistsForURLScheme(StringView) const final;
-    String generatedMIMETypeForURLScheme(StringView) const final;
-
-    void frameLoadCompleted() final;
-    void restoreViewState() final;
-    void provisionalLoadStarted() final;
-    void didFinishLoad() final;
-    void prepareForDataSourceReplacement() final;
-
-    void updateCachedDocumentLoader(DocumentLoader&) final;
-    void setTitle(const StringWithDirection&, const URL&) final;
+    void dispatchDidFailLoading(DocumentLoader*, ResourceLoaderIdentifier, const ResourceError&) override;
+    bool dispatchDidLoadResourceFromMemoryCache(DocumentLoader*, const ResourceRequest&, const ResourceResponse&, int) override;
+
+    void dispatchDidDispatchOnloadEvents() override;
+    void dispatchDidReceiveServerRedirectForProvisionalLoad() override;
+    void dispatchDidCancelClientRedirect() override;
+    void dispatchWillPerformClientRedirect(const URL&, double, WallTime, LockBackForwardList) override;
+    void dispatchDidChangeLocationWithinPage() override;
+    void dispatchDidPushStateWithinPage() override;
+    void dispatchDidReplaceStateWithinPage() override;
+    void dispatchDidPopStateWithinPage() override;
+    void dispatchWillClose() override;
+    void dispatchDidStartProvisionalLoad() override;
+    void dispatchDidReceiveTitle(const StringWithDirection&) override;
+    void dispatchDidCommitLoad(std::optional<HasInsecureContent>, std::optional<UsedLegacyTLS>, std::optional<WasPrivateRelayed>) override;
+    void dispatchDidFailProvisionalLoad(const ResourceError&, WillContinueLoading, WillInternallyHandleFailure) override;
+    void dispatchDidFailLoad(const ResourceError&) override;
+    void dispatchDidFinishDocumentLoad() override;
+    void dispatchDidFinishLoad() override;
+    void dispatchDidReachLayoutMilestone(OptionSet<LayoutMilestone>) override;
+    void dispatchDidReachVisuallyNonEmptyState() override;
+
+    LocalFrame* dispatchCreatePage(const NavigationAction&, NewFrameOpenerPolicy) override;
+    void dispatchShow() override;
+
+    void dispatchDecidePolicyForResponse(const ResourceResponse&, const ResourceRequest&, const String&, FramePolicyFunction&&) override;
+    void dispatchDecidePolicyForNewWindowAction(const NavigationAction&, const ResourceRequest&, FormState*, const String&, std::optional<HitTestResult>&&, FramePolicyFunction&&) override;
+    void dispatchDecidePolicyForNavigationAction(const NavigationAction&, const ResourceRequest&, const ResourceResponse& redirectResponse, FormState*, const String&, std::optional<NavigationIdentifier>, std::optional<HitTestResult>&&, bool, NavigationUpgradeToHTTPSBehavior, SandboxFlags, PolicyDecisionMode, FramePolicyFunction&&) override;
+    void updateSandboxFlags(SandboxFlags) override;
+    void updateOpener(std::optional<FrameIdentifier>) override;
+    void setPrinting(bool, FloatSize, FloatSize, float, AdjustViewSize) override;
+    void cancelPolicyCheck() override;
+
+    void dispatchUnableToImplementPolicy(const ResourceError&) override;
+
+    void dispatchWillSendSubmitEvent(Ref<FormState>&&) override;
+    void dispatchWillSubmitForm(FormState&, URL&& requestURL, String&& method, CompletionHandler<void()>&&) override;
+
+    void revertToProvisionalState(DocumentLoader*) override;
+    void setMainDocumentError(DocumentLoader*, const ResourceError&) override;
+
+    void setMainFrameDocumentReady(bool) override;
+
+    void startDownload(const ResourceRequest&, const String&, FromDownloadAttribute = FromDownloadAttribute::No) override;
+
+    void willChangeTitle(DocumentLoader*) override;
+    void didChangeTitle(DocumentLoader*) override;
+
+    void willReplaceMultipartContent() override;
+    void didReplaceMultipartContent() override;
+
+    void committedLoad(DocumentLoader*, const SharedBuffer&) override;
+    void finishedLoading(DocumentLoader*) override;
+
+    void loadStorageAccessQuirksIfNeeded() override;
+
+    bool shouldFallBack(const ResourceError&) const override;
+
+    bool canHandleRequest(const ResourceRequest&) const override;
+    bool canShowMIMEType(const String&) const override;
+    bool canShowMIMETypeAsHTML(const String&) const override;
+    bool representationExistsForURLScheme(StringView) const override;
+    String generatedMIMETypeForURLScheme(StringView) const override;
+
+    void frameLoadCompleted() override;
+    void restoreViewState() override;
+    void provisionalLoadStarted() override;
+    void didFinishLoad() override;
+    void prepareForDataSourceReplacement() override;
+
+    void updateCachedDocumentLoader(DocumentLoader&) override;
+    void setTitle(const StringWithDirection&, const URL&) override;
 
     String userAgent(const URL&) const override;
 
-    void savePlatformDataToCachedFrame(CachedFrame*) final;
-    void transitionToCommittedFromCachedFrame(CachedFrame*) final;
+    void savePlatformDataToCachedFrame(CachedFrame*) override;
+    void transitionToCommittedFromCachedFrame(CachedFrame*) override;
 #if PLATFORM(IOS_FAMILY)
-    void didRestoreFrameHierarchyForCachedFrame() final;
+    void didRestoreFrameHierarchyForCachedFrame() override;
 #endif
-    void transitionToCommittedForNewPage(InitializingIframe) final;
+    void transitionToCommittedForNewPage(InitializingIframe) override;
 
-    void didRestoreFromBackForwardCache() final;
+    void didRestoreFromBackForwardCache() override;
 
-    void updateGlobalHistory() final;
-    void updateGlobalHistoryRedirectLinks() final;
-    ShouldGoToHistoryItem shouldGoToHistoryItem(HistoryItem&, IsSameDocumentNavigation, ProcessSwapDisposition) const final;
-    bool supportsAsyncShouldGoToHistoryItem() const final;
-    void shouldGoToHistoryItemAsync(HistoryItem&, CompletionHandler<void(ShouldGoToHistoryItem)>&&) const final;
+    void updateGlobalHistory() override;
+    void updateGlobalHistoryRedirectLinks() override;
+    ShouldGoToHistoryItem shouldGoToHistoryItem(HistoryItem&, IsSameDocumentNavigation, ProcessSwapDisposition) const override;
+    bool supportsAsyncShouldGoToHistoryItem() const override;
+    void shouldGoToHistoryItemAsync(HistoryItem&, CompletionHandler<void(ShouldGoToHistoryItem)>&&) const override;
 
-    void saveViewStateToItem(HistoryItem&) final;
-    bool canCachePage() const final;
-    RefPtr<LocalFrame> createFrame(const AtomString&, HTMLFrameOwnerElement&) final;
-    RefPtr<Widget> createPlugin(HTMLPlugInElement&, const URL&, const Vector<AtomString>&, const Vector<AtomString>&, const String&, bool) final;
+    void saveViewStateToItem(HistoryItem&) override;
+    bool canCachePage() const override;
+    RefPtr<LocalFrame> createFrame(const AtomString&, HTMLFrameOwnerElement&) override;
+    RefPtr<Widget> createPlugin(HTMLPlugInElement&, const URL&, const Vector<AtomString>&, const Vector<AtomString>&, const String&, bool) override;
 
-    ObjectContentType objectContentType(const URL&, const String&) final;
-    AtomString overrideMediaType() const final;
+    ObjectContentType objectContentType(const URL&, const String&) override;
+    AtomString overrideMediaType() const override;
 
-    void redirectDataToPlugin(Widget&) final;
-    void dispatchDidClearWindowObjectInWorld(DOMWrapperWorld&) final;
+    void redirectDataToPlugin(Widget&) override;
+    void dispatchDidClearWindowObjectInWorld(DOMWrapperWorld&) override;
 
 #if PLATFORM(COCOA)
-    RemoteAXObjectRef accessibilityRemoteObject() final;
-    IntPoint accessibilityRemoteFrameOffset() final;
+    RemoteAXObjectRef accessibilityRemoteObject() override;
+    IntPoint accessibilityRemoteFrameOffset() override;
 #if ENABLE(ACCESSIBILITY_ISOLATED_TREE)
-    void setIsolatedTree(Ref<WebCore::AXIsolatedTree>&&) final;
-    RefPtr<WebCore::AXIsolatedTree> isolatedTree() const final;
+    void setIsolatedTree(Ref<WebCore::AXIsolatedTree>&&) override;
+    RefPtr<WebCore::AXIsolatedTree> isolatedTree() const override;
 #endif
-    void willCacheResponse(DocumentLoader*, ResourceLoaderIdentifier, NSCachedURLResponse *, CompletionHandler<void(NSCachedURLResponse *)>&&) const final;
+    void willCacheResponse(DocumentLoader*, ResourceLoaderIdentifier, NSCachedURLResponse *, CompletionHandler<void(NSCachedURLResponse *)>&&) const override;
 #endif
 
-    Ref<FrameNetworkingContext> createNetworkingContext() final;
+    Ref<FrameNetworkingContext> createNetworkingContext() override;
 
     bool isEmptyFrameLoaderClient() const override;
-    void prefetchDNS(const String&) final;
-    void sendH2Ping(const URL&, CompletionHandler<void(Expected<Seconds, ResourceError>&&)>&&) final;
+    void prefetchDNS(const String&) override;
+    void sendH2Ping(const URL&, CompletionHandler<void(Expected<Seconds, ResourceError>&&)>&&) override;
 
 #if USE(QUICK_LOOK)
-    RefPtr<LegacyPreviewLoaderClient> createPreviewLoaderClient(const String&, const String&) final;
+    RefPtr<LegacyPreviewLoaderClient> createPreviewLoaderClient(const String&, const String&) override;
 #endif
 
-    bool hasFrameSpecificStorageAccess() final;
-    void revokeFrameSpecificStorageAccess() final;
+    bool hasFrameSpecificStorageAccess() override;
+    void revokeFrameSpecificStorageAccess() override;
 
-    void dispatchLoadEventToOwnerElementInAnotherProcess() final;
+    void dispatchLoadEventToOwnerElementInAnotherProcess() override;
 
-    RefPtr<HistoryItem> createHistoryItemTree(bool, BackForwardItemIdentifier) const final;
+    RefPtr<HistoryItem> createHistoryItemTree(bool, BackForwardItemIdentifier) const override;
 };
 
 } // namespace WebCore
